pageserver/src/restore_local_repo.rs

```diff
@@ -241,3 +242,133 @@ pub fn import_timeline_wal(walpath: &Path, timeline: &dyn Timeline, startpoint:
     info!("reached end of WAL at {}", last_lsn);
     Ok(())
 }
+
+///
+/// Helper function to parse a WAL record and call the Timeline's PUT functions for all the
+/// relations/pages that the record affects.
+///
+pub fn save_decoded_record(
+    timeline: &dyn Timeline,
+    decoded: DecodedWALRecord,
+    recdata: Bytes,
+    lsn: Lsn,
+) -> Result<()> {
+    // Figure out which blocks the record applies to, and "put" a separate copy
+    // of the record for each block.
+    for blk in decoded.blocks.iter() {
+        let tag = BufferTag {
+            rel: RelTag {
+                spcnode: blk.rnode_spcnode,
+                dbnode: blk.rnode_dbnode,
+                relnode: blk.rnode_relnode,
+                forknum: blk.forknum as u8,
+            },
+            blknum: blk.blkno,
+        };
```

Comment:
Seems like creating `RelTag` & `BufferTag` could use helper functions to avoid so much boilerplate. E.g. `RelTag::from_decoded_bkp_block(blk)`, maybe with less verbose name :)

Might be offtopic for this review though, feel free to ignore for now.